src/main/java/uk/gov/companieshouse/officerfiling/api/validation/OfficerTerminationValidator.java

```diff
@@ -54,16 +55,32 @@ public ApiErrors validate(HttpServletRequest request, OfficerFilingDto dto, Stri
 
         // Retrieve data objects required for the validation process
         final Transaction transaction = transactionService.getTransaction(transId, passthroughHeader);
-        final AppointmentFullRecordAPI companyAppointment = companyAppointmentService.getCompanyAppointment(transaction.getCompanyNumber(), dto.getReferenceAppointmentId(), passthroughHeader);
+        final AppointmentFullRecordAPI companyAppointment = validateOfficerIsIdentified(request, errorList, dto, transaction, passthroughHeader);
         final CompanyProfileApi companyProfile = companyProfileService.getCompanyProfile(transId, transaction.getCompanyNumber(), passthroughHeader);
 
-        // Perform validation
-        validateMinResignationDate(request, errorList, dto);
-        validateTerminationDateAfterIncorporationDate(request, errorList, dto, companyProfile, companyAppointment);
+        if (companyAppointment != null) {
```

Comment:
I think it would be better to have an else and return Apierrors(errorlist) Because at the moment its fine, but someone could come along and add new validation outside of your if statement, when in reality we want to fail now as there is no point in continuing.